prov/efa/src/efa_user_info.c

```diff
@@ -571,51 +571,36 @@ int efa_getinfo(uint32_t version, const char *node,
 {
 	struct fi_info *dgram_info_list, *rdm_info_list;
 	int err;
-	static bool shm_info_initialized = false;
-
-	/*
-	 * efa_shm_info_initialize() initializes the global variable g_shm_info.
-	 * Ideally it should be called during provider initialization. However,
-	 * At the time of EFA provider initialization, shm provider has not been
-	 * initialized yet, therefore g_shm_info cannot be initialized. As a workaround,
-	 * we initialize g_shm_info when the rxr_getinfo() is called 1st time,
-	 * at this point all the providers have been initialized.
-	 */
-	if (!shm_info_initialized) {
-		efa_shm_info_initialize(hints);
-		shm_info_initialized = true;
-	}
 
 	if (hints && hints->ep_attr && hints->ep_attr->type == FI_EP_DGRAM) {
 		err = efa_user_info_get_dgram(version, node, service, flags, hints, info);
 		if (err)
-			goto error;
+			return err;
```

Comment:
Why not just `return efa_user_info_get_dgram(xxx)`?

Reply:
It seems you are reading an older revision? I already made  the change and I could not see this in file changes now